### PR TITLE
refactor(es6): Upgrade AbstractPlugin to es6

### PR DESCRIPTION
**Pull request by chicoxyzzy**
**What kind of change does this PR introduce?**

Refactor

**Did you add tests for your changes?**

Existing Tests are passing

**If relevant, link to documentation update:**

N/A

**Summary**

update AbstractPlugin to ES6

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: lib/AbstractPlugin.js [hunk 1]
```diff
@@ -2,21 +2,26 @@
...
+				super(plugins);
+			}
+		};
+	}
 
-AbstractPlugin.create = function(plugins) {
-	function Plugin() {
-		AbstractPlugin.call(this, plugins);
+	constructor(plugins) {
+		this._plugins = plugins || {};
```
Comment (Kovensky) [inline lib/AbstractPlugin.js]: It'd be nice to convert this into a `Map`, but that'd be a larger scale refactoring

Comment (TheLarkInn) [inline lib/AbstractPlugin.js]: Let's put an issue in for someone to tackle if they want to down the road.

Comment (chicoxyzzy) [inline lib/AbstractPlugin.js]: yes, I think this could be done in another PR

**Comment by TheLarkInn** [comment]
Thank you :-D

**Pull request merged**